torch/_inductor/fx_passes/split_cat.py

```diff
@@ -1174,6 +1174,10 @@ def merge_getitem_cat(match: Match, split_sections: List[int], dim: int):
                 indices.append(arg.args[1])
             # indices may not be necessarily sorted, we sort them first
             indices.sort()
+            # the gettitems to be merged must be consecutive, otherwise
+            # returned sliced tensor could be wrong
+            if indices[len(indices)-1] - indices[0] + 1 != len(indices):
```

Comment:
Can there be duplicated indices? If so, e.g. `[0, 1, 1, 1, 1, 5]` would pass this test but the indices are not consecutive. Seems safer to do `indices == list(arange(len(indices)))`.

Reply:
Also nit: can do `indices[-1]` instead of `indices[len(indices)-1]`. But @peterbell10's suggestion is better, I believe.